File: src/test/java/io/phdata/snowpark/metrics/SessionConfigBase.java
package io.phdata.snowpark.metrics;

import com.snowflake.snowpark.Session;
import io.phdata.snowpark.helpers.Config;
import io.phdata.snowpark.helpers.ConfigBuilder$;
import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;
import org.scalatest.junit.JUnitSuite;

public class SessionConfigBase extends JUnitSuite {

    public static Config config;
    public static Session session;

    @BeforeClass
    public static void setupClass() {
        config = ConfigBuilder$.MODULE$.build((new String[]{}));
        session = Session.builder().configs(config.getSnowparkConnectionProperties()).create();
    }

    @AfterClass
    public static void tearDown() {
        if (session != null) {
            session.close();
        }
    }

    @Test
    public void dummy() { Assert.assertTrue(true);}


    protected Session getSession() { return session; }
}

File: src/test/scala/io/phdata/snowpark/metrics/SessionConfigBase.scala
package io.phdata.snowpark.metrics

import com.snowflake.snowpark.Session
import io.phdata.snowpark.helpers.{Config, ConfigBuilder}
import io.phdata.snowpark.metrics.SessionConfigBase.session
import org.junit.AfterClass
import org.junit.BeforeClass


object SessionConfigBase {
  var config: Option[Config] = None
  var session: Option[Session] = None

  @BeforeClass
  def setupClass(): Unit = {
    config = Some(ConfigBuilder.build(Array()))
    session = Some(Session.builder.configs(config.get.getSnowparkConnectionProperties).create)
  }

  @AfterClass
  def tearDown(): Unit = {
   session.foreach(_.close())
  }

}

class SessionConfigBase {
  protected def getSession: Session = session.get
}